shared/js/toc.jsx

```diff
@@ -0,0 +1,772 @@
+    },
+
+    /**
+     * Checks if current room is active.
+     *
+     * @return {Boolean}
+     */
+    _roomIsActive: function() {
+      return this.state.roomState === ROOM_STATES.JOINED ||
+             this.state.roomState === ROOM_STATES.SESSION_CONNECTED ||
+             this.state.roomState === ROOM_STATES.HAS_PARTICIPANTS;
+    },
+
+    render: function() {
+      var displayScreenShare = !!(this.state.receivingScreenShare ||
+        this.props.screenSharePosterUrl);
+
+      return (
+
+        <div className="room-conversation-wrapper standalone-room-wrapper">
+          <TableOfContentView
+            activeRoomStore={this.props.activeRoomStore}
+            dispatcher={this.props.dispatcher}
+            isDesktop={true}
+            isScreenShareActive={displayScreenShare} />
+
+          <sharedViews.ScreenShareView
+            cursorStore={this.props.cursorStore}
+            dispatcher={this.props.dispatcher}
+            displayScreenShare={displayScreenShare}
+            isScreenShareLoading={this._isScreenShareLoading()}
+            screenShareMediaElement={this.state.screenShareMediaElement}
+            screenSharePosterUrl={this.props.screenSharePosterUrl}
+            screenSharingPaused={this.state.streamPaused} />
+
+          <SidebarView
+            activeRoomStore={this.props.activeRoomStore}
+            audio={{ enabled: !this.state.audioMuted,
+                     visible: this._roomIsActive() }}
+            dispatcher={this.props.dispatcher}
+            isFirefox={this.props.isFirefox}
+            leaveRoom={this.leaveRoom}
+            video={{ enabled: !this.state.videoMuted,
+                     visible: this._roomIsActive() }} />
+        </div>
+      );
+    }
+  });
+
+  /* This is a fork of the StandlaoneRoomControllerView */
+  var RoomControllerView = React.createClass({
+    mixins: [
+      loop.store.StoreMixin("activeRoomStore")
+    ],
+
+    propTypes: {
+      cursorStore: React.PropTypes.instanceOf(loop.store.RemoteCursorStore).isRequired,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      isFirefox: React.PropTypes.bool.isRequired
+    },
+
+    getInitialState: function() {
+      return this.getStoreState();
+    },
+
+    render: function() {
+      // If we don't know yet, don't display anything.
+      if (this.state.userAgentHandlesRoom === undefined) {
+        return null;
+      }
+
+      // if (this.state.userAgentHandlesRoom) {
+      //   return (
+      //     <StandaloneHandleUserAgentView
+      //       dispatcher={this.props.dispatcher} />
+      //   );
+      // }
+
+      // XXX force into HAS_PARTICIPANTS mode...
+      return (
+        <RoomView
+          activeRoomStore={this.getStore()}
+          cursorStore={this.props.cursorStore}
+          dispatcher={this.props.dispatcher}
+          isFirefox={this.props.isFirefox} />
+      );
+    }
+  });
+
+  var SidebarView = React.createClass({
```

Comment:
It was forked from StandaloneRoomView since the sidebar need to have the same data as the toc. I agree with your comment, it needs to be shared and perhaps we could do a bit clean up of this component before landing into akita. e.g. Remove introSeen state